File: frontend/types/agent.ts
export interface Agent {
    id: string;
    business_id: string;
    name: string;
    description?: string;
    system_prompt: string;
    language: string;
    voice_id: string;
    stability: number;
    similarity_boost: number;
    created_at: string;
    updated_at: string;
}

export interface AgentCreate {
    business_id?: string;
    name: string;
    description?: string;
    system_prompt: string;
    language?: string;
    voice_id?: string;
    stability?: number;
    similarity_boost?: number;
}

export interface AgentUpdate {
    name?: string;
    description?: string;
    system_prompt?: string;
    language?: string;
    voice_id?: string;
    stability?: number;
    similarity_boost?: number;
}

File: backend/app/models/agent.py
from sqlalchemy import Column, String, ForeignKey, DateTime, Text, Float
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
import uuid
from datetime import datetime, timezone

from app.db.base import Base


class Agent(Base):

    __tablename__ = "agents"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    business_id = Column(UUID(as_uuid=True), ForeignKey("businesses.id"), nullable=False)

    name = Column(String(255), nullable=False)
    description = Column(Text, nullable=True)
    
    # AI configuration
    system_prompt = Column(Text, nullable=False)
    language = Column(String(50), default="en")
    voice_id = Column(String(100), default="alloy")
    
    # ElevenLabs style settings (optional but good for "enhanced experience")
    stability = Column(Float, default=0.5)
    similarity_boost = Column(Float, default=0.75)
    
    # Timestamps
    created_at = Column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc))
    updated_at = Column(
        DateTime(timezone=True),
        default=lambda: datetime.now(timezone.utc),
        onupdate=lambda: datetime.now(timezone.utc),
    )

    # Relationships
    business = relationship("Business", backref="agents")
    campaigns = relationship("Campaign", back_populates="agent")
